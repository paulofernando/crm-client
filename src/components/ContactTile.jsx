import React from 'react';
import './Style.css';


class ContactTile extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div className="contactCard">
                <div className={`${ this.props.contact.caseRole.toLowerCase() }`}>
                    <h3>{this.props.contact.firstName} {this.props.contact.lastName}</h3>
                    <spam>{this.props.contact.email}</spam>
                    <p>#{this.props.contact.courtCase.id}</p>
                </div>
            </div>
        );
    }
}

export default ContactTile;