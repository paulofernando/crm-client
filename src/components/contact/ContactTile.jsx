import React from 'react';
import { Link } from '@reach/router';

import '../ComponentStyle.css';

const ContactTile = (props) => {     
    return (
        <Link style={{ textDecoration: 'none' }} to={`/contact/${props.contact.id}`}>
            <div className="smallContactCard">
                <div className={`${props.contact.caseRole.toLowerCase()}`}>
                    {props.contact.firstName}
                </div>
            </div>
        </Link>
    );    
}

export default ContactTile;