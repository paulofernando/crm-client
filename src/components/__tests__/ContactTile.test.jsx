import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ContactTile from '../contact/ContactTile';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

// -------------------------------

const ContactTileComponent = (<ContactTile contact={contact} />)

// -------------------------------

describe('ContactTile', () => {

    it('renders correctly', () => {
        const contact = {
            firstName: "John",
            lastName: "Smith",
            caseRole: "Judge",
            email: "js@js.com",
            courtCaseId: 1
        }

        const wrapper = renderer
            .create(ContactTileComponent)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});