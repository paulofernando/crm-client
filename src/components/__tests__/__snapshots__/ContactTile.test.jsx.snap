// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTile renders correctly 1`] = `
<a
  href="/contact/undefined"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <div
    className="smallContactCard"
  >
    <div
      className="judge"
    >
      John
    </div>
  </div>
</a>
`;
